packages/catalog-process/src/services/catalogService.ts

```diff
@@ -444,7 +453,12 @@ export function catalogServiceBuilder(
         descriptors: [],
         createdAt: creationDate,
         riskAnalysis: [],
-        isSignalHubEnabled: seed.isSignalHubEnabled,
+        isSignalHubEnabled: config.featureFlagSignalhubWhitelist
+          ? isTenantInSignalHubWhitelist(
+              authData.organizationId,
+              seed.isSignalHubEnabled
+            )
+          : seed.isSignalHubEnabled,
```

Comment:
```suggestion
        isSignalHubEnabled:
          seed.isSignalHubEnabled &&
          isTenantInSignalHubWhitelist(authData.organizationId),
```
How about implementing it like this? In this case we check the flag only if `isSignalHubEnable` is true, otherwise there is no need. See my suggestion on `isTenantInSignalHubWhitelist`